pkg/traces/traces_test.go

```diff
@@ -78,7 +78,7 @@ configs:
       tls_config:
           insecure_skip_verify: true
   spanmetrics:
-    handler_endpoint: 0.0.0.0:9090
+    handler_endpoint: 0.0.0.0:8888
```

Comment:
Correcting myself, I think the problem was already there, but it materialized now since we were forcing the grpc port to 9090 (was not previously working when not setting it). Will be fixed now that we are back to random grpc port by default but there seems to be a server not being closed correctly. Will look for it

Reply:
Found it, fixed in https://github.com/grafana/agent/pull/3888/commits/109a37f917aa505b3aecf05fea606a33b21acf0f